scripts/core/editor3/components/DraftSpellchecker.jsx

```diff
@@ -0,0 +1,240 @@
+/**
+ * @ngdoc React
+ * @module superdesk.core.editor3
+ * @name DraftSpellchecker
+ * @param {String} language the current spellchecker language
+ * @description check the text and shos the inline spellchecker context menu
+ */
+
+import React from 'react';
+import {EditorState, Modifier} from 'draft-js';
+import {InlineSpellcheckMenu} from '../components/InlineSpellcheckMenu';
+
+export class DraftSpellchecker {
+    constructor(language) {
+        this.language = language;
+        this.show = false;
+
+        this.registerContextMenuCallbacks = this.registerContextMenuCallbacks.bind(this);
+        this.registerDictionaryCallbacks = this.registerDictionaryCallbacks.bind(this);
+
+        this.fullSpellcheck = this.fullSpellcheck.bind(this);
+        this.currentBlockSpellcheck = this.currentBlockSpellcheck.bind(this);
+
+        this.showContextMenu = this.showContextMenu.bind(this);
+        this.getContextMenu = this.getContextMenu.bind(this);
+
+        this.ignoreWord = this.ignoreWord.bind(this);
+        this.addWord = this.addWord.bind(this);
+        this.replaceWord = this.replaceWord.bind(this);
+    }
+
+    /** Register the context menu related actions */
+    registerContextMenuCallbacks(onOpenContextMenu, onCloseContextMenu, onReplaceWord) {
+        this.onOpenContextMenu = onOpenContextMenu;
```

Comment:
I would initialise all of these listeners in the constructor as `null`, to be able to document them and to offer a better overview of the attributes of this class. It makes it easier to understand the behaviour if all the attributes are listed in one place.